analyzer/src/main/kotlin/managers/Bower.kt

```diff
@@ -141,13 +141,69 @@ class Bower(analyzerConfig: AnalyzerConfiguration, repoConfig: RepositoryConfigu
             return result
         }
 
+        private fun hasCompleteDependencies(node: JsonNode, scopeName: String): Boolean {
+            val dependencyKeys = node["dependencies"].fieldNamesOrEmpty().asSequence().toSet()
+            val dependencyRefKeys = node["pkgMeta"][scopeName].fieldNamesOrEmpty().asSequence().toSet()
+
+            return dependencyKeys.containsAll(dependencyRefKeys)
+        }
+
+        private fun dependencyKeyOf(node: JsonNode): String? {
+            // As non-null dependency keys are supposed to as equivalence relation for parsing 'missing' nodes,
```

Comment:
I can't parse the first part of this sentence. What do you mean with "equivalence relation"?

Reply:
equiavalent(a,b) <==> dependencyKeyOf(a) == dependencyKeyOf(b) != null

Reply:
that is, this function defines whether a node can be replaced with another